### PR TITLE
Deprecate unneeded meta-headers

### PR DESCRIPTION
**Pull request by rabaho**
Deprecated the following:
pcl/common/boost.h, pcl/sample_consensus/boost.h, pcl/octree/boost.h, pcl/recognition/boost, pcl/features/boost, pcl/features/eigen.h, and correspondence_rejection_median_distance.hpp

This is tied to [https://github.com/PointCloudLibrary/pcl/issues/4290](4290)

### CODE REVIEW

**Review by kunaltyagi — APPROVED** [review]
This change-set was surprisingly smaller than I anticipated

**Comment by mvieth** [comment]
I looked a bit further into the registration module, there are more hpp files that are basically empty: icp_nl.hpp and many of the correspondence_rejection_*.hpp files. I guess we can deprecate them all @kunaltyagi ?

It would be nice if there was a deprecation notice like "Please include the needed [eigen/boost] headers directly" where appropriate.

**Comment by rabaho** [comment]
@mvieth I noticed those empty headers as well, I wasn't sure if they needed to be deprecated. We can, if you guys think they should be deprecated.

**Comment by mvieth** [comment]
I would say yes, deprecate them. I don't see a reason why we need these empty headers, which are also unnecessarily included by the .h files.

File: features/include/pcl/features/boundary.h [hunk 1]
```diff
@@ -40,7 +40,6 @@
 
 #pragma once
 
-#include <pcl/features/eigen.h>
```
Comment (mvieth) [inline features/include/pcl/features/boundary.h:43]: Is it save to just remove this? The checks have passed, so there is nothing obvious missing, but that file contains `#include <Eigen/StdVector>`, which might not give errors when it is missing, but might still cause subtle problems? I think the safest way is to replace the includes you have removed (eigen.h / boost.h, more below)  with their respective contents (unless you are sure that a file does not need a specific include).

Comment (larshg) [reply features/include/pcl/features/boundary.h:43]: I guess it should be verfied what headers are required and not just add all. 
If we just move all, we really haven't gained anything?

Comment (kunaltyagi) [reply features/include/pcl/features/boundary.h:43]: This file (or the impl or the cpp) don't have any `vector<Eigen::X>` so I think it's safe to remove the include

Comment (rabaho) [reply features/include/pcl/features/boundary.h:43]: I believe their removal is safe. I checked the code and didn't find any usage for them.

**Review by mvieth — APPROVED** [review]

File: registration/include/pcl/registration/correspondence_rejection_var_trimmed.h [hunk 2]
```diff
@@ -259,6 +259,4 @@ class PCL_EXPORTS CorrespondenceRejectorVarTrimmed : public CorrespondenceReject
   optimizeInlierRatio(std::vector<double>& dists) const;
 };
 } // namespace registration
-} // namespace pcl
-
-#include <pcl/registration/impl/correspondence_rejection_var_trimmed.hpp>
+} // namespace pcl
```
Comment (larshg) [inline registration/include/pcl/registration/correspondence_rejection_var_trimmed.h:262]: Missing empty line?

**Review by larshg — APPROVED** [review]

**Pull request merged**